source/agora/consensus/EnrollmentManager.d

```diff
@@ -104,6 +107,17 @@ public class EnrollmentManager
             signature_noise = deserializeFull!Pair(row.peek!(ubyte[])(0));
             break;
         }
+
+        // load preimages
+        results = this.db.execute("SELECT val FROM node_enroll_data " ~
+            "WHERE key = ?", "preimages");
+
+        foreach (row; results)
```

Comment:
I think maybe this would work too:

```D
if (!results.empty)
    this.preimages = results.oneValue!(ubyte[]).deserializeFull!(Hash[]);
```

Could be worth trying.